### ISSUE TITLE
update desisim.targets.get_targets() to use the new template simulations code

### ISSUE THREAD

**Issue by moustakas** [desihub/desisim#32]
Build the templates on-the-fly by calling the relevant functions in desisim.templates rather than using the static (on-disk) templates.

**Comment by moustakas** [comment]
@sbailey In get_targets() you build the wavelength vector thusly:
```
    wavemin = desimodel.io.load_throughput('b').wavemin
    wavemax = desimodel.io.load_throughput('z').wavemax
    dw = 0.2
    wave = np.arange(round(wavemin, 1), wavemax, dw)
```

I'm unclear about the call to "round()".  Don't we either want to use the exact value or, at the very least, make this a floor?

In desisim.templates I do
```
       npix = (maxwave-minwave)/cdelt+1
       wave = np.linspace(minwave,maxwave,npix) 
```
which gives a close but not an identical output wavelength array (with maxwave=wavemax, minwave=wavemin, and cdelt=dw, obviously).

Thoughts?

**Comment by sbailey** [comment]
I think my reasoning was that I wanted a 0.2 Angstrom grid aligned to an even tenth of an Angstrom starting point.  The exact phase of the grid isn't critical, and I didn't care if I was a fraction of an Angstrom on or off the CCD at the very edge.  Your method guarantees that minwave and maxwave are exactly included, but only gives a spacing of cdelt if (maxwave-minwave)/cdelt is an integer (within floating point rounding, which it may not be).  I prioritized having a spacing of cdelt, at the expense of missing maxwave by a fraction of a cdelt if (maxwave-minwave)/cdelt isn't an integer.  It probably doesn't matter in any practical way for a cdelt of 0.2 over thousands of Angstroms, but I just like to see wavelength grids with human readable number of digits when I'm looking at spectra unless there is a particularly good reason not to.

Note: the wavelength grid of these files is the grid for projecting photons onto the CCD, so it is important for it to be less than a pixel size, and linear rather than log spacing.

**Comment by moustakas** [comment]
In bin/newexp-desi (actually desisim.targets.get_targets) what default redshift and magnitude priors do we want for each object type?  This is what I have so far:
```
QSO: g=[21,23], redshift=[0.5,4]
ELG: r=[21,23.5], redshift=[0.6,1.6]
LRG: z=[19,20.5], redshift=[0.5,1.1]
STAR: r=[18,23.5] (0+/-200 km/s Gaussian radial velocity jitter included by default)
STD: r=[16,19] (0+/-200 km/s Gaussian radial velocity jitter included by default)
```
These are pretty faint magnitude limits, so the spectra are going to be pretty noisy for many objects. This is a choice, but should I make the defaults go a bit brighter while we're in testing mode so we can get see some glorious successes when we take the simulations through redshift-fitting?  These priors are obviously fully customizable.

**Comment by sbailey** [comment]
These are pretty close to the [baseline target selection cuts](https://desi.lbl.gov/trac/wiki/TargetSelectionWG/TargetSelection), so let's keep ourselves honest with faint sources.  If we really have trouble with them, we'll have to make adjustments to TS and/or exposure times and we might as well learn that now.

Small differences with baseline TS cuts:
LRG z < 20.56 vs. 20.5
ELG r < 23.4 vs. 23.5
QSO cut on r<23 vs. 21 < g < 23

**Comment by moustakas** [comment]
OK, honesty = good.  I'm going to keep the default (round) values because when we actually sit down to quantify the redshift success we're going to want push (a bit) fainter than the nominal limits.

I've also added D4000 as a "truth table" output to the ELG and LRG spectral classes, as redshift success will/should correlate with these quantities, especially for LRGs.

**Comment by moustakas** [comment]
Fixed in PR #34, which was merged to master.  Closing
